### DISCUSSION TITLE
Running into the error "Missing file index.html. Please try again." when trying to access the `skipadmin` page

### DISCUSSION THREAD

**Discussion by chris-schmitz** [mozilla/hubs#3082] [General Questions & Discussion] [answered]
I'm trying to setup a local version of recticulum and hubs instance so that my team and I can add in custom websocket messaging to our hubs application and simplify our dev environments. 

I pulled down the reticulum and hubs codebases and walked through the reticulum readme. I hit a snag with accessing the reticulum/hubs server because of a snag in the reticulum readme (the instructions for the `hosts` file have you add a host entry that is `domain name then ip` but it should be `ip then domain name`):

<img width="317" alt="Screen Shot 2020-09-28 at 12 10 09 PM" src="https://user-images.githubusercontent.com/2437424/94476134-abdbdd80-0195-11eb-99b0-983f858332b7.png">

Once that was corrected I was able to hit the `https://hubs.local:4000?skipadmin` address and see it reflected in the reticulum console output, but I do not get the magic link in the console that would normally go out via email:

![skipadmin](https://user-images.githubusercontent.com/2437424/94476876-b5197a00-0196-11eb-8177-7585ba9ea47f.gif)

And the error I get in the browser is `Missing file index.html. Please try again.`. 

And the debug output I get in the reticulum console is:

```
[debug] Processing with RetWeb.PageController
  Parameters: [UNFETCHED]
  Pipelines: [:secure_headers, :parsed_body, :browser]
[debug] QUERY OK source="app_configs" db=0.3ms idle=9015.9ms
SELECT a0."app_config_id", a0."key", a0."value", a0."owned_file_id", a0."inserted_at", a0."updated_at" FROM "ret0"."app_configs" AS a0 WHERE (a0."key" = $1) ["features|default_room_id"]
[debug] QUERY OK source="app_configs" db=0.3ms idle=8016.4ms
SELECT a0."app_config_id", a0."key", a0."value", a0."owned_file_id", a0."inserted_at", a0."updated_at" FROM "ret0"."app_configs" AS a0 []
[info] Sent 503 in 10003ms
```

Any idea of what could be causing this or where I should start digging in?

**Comment by robinkwilson** [comment]
1. Can you try navigating to `reticulum/assets` directory and running `npm ci`? (https://github.com/mozilla/reticulum/tree/master/assets)
2. Make sure hubs is running locally via `npm run local`, then navigate to https://hubs.local:8080 to accept the self signed certificate (I believe is the local port it's running on)
3. Delete local storage for https://hubs.local:4000
4. Ensure you have postgres running specified in the reticulum readme.

Good luck!

Also just to reiterate, my experience before I got Hubs Dev credentials, I wasn't able to run a reticulum locally from the reticulum README instructions. 
@netpro2k may know more, but I believe it's because you're still using the Hubs Developer dialog server which you'd need to be an admin to access. 

That being said, would LOVE to see this work!

**Reply by robinkwilson** [reply]
Marking this as answered for now, so we can remove it from the pipeline! @ me when you respond!

**Reply by chris-schmitz** [reply]
Hey Robin,

Sorry I missed this message the other day. I'll hop in and give it a try this afternoon and let you know what I find. Thanks for the reply!

**Reply by chris-schmitz** [reply]
Yeah @robin-k-wilson, still no go. I Tried the steps you noted and I'm still getting errors trying to use skipadmin on the reticulum site, I don't see an invite in the `hub_invites`table (I figured I may be able to pull the invite info from there, I forgot to show in the gif below that if I re-query the invite table it's still blank), and when I try to use my local hubs it does look like it's talking to my local reticulum but i'm still spinning and getting errors. 

![still snagging](https://user-images.githubusercontent.com/2437424/95103797-b13ba980-06fa-11eb-9bf2-6bb93076777c.gif)

The hard part that I'm bumping up against is that with little context on the project and being new to elixir and phoenix I'm a bit lost trying to debug. 

We talked internally and we're going to put a hold on the local dev setup for now and use workarounds for the custom messaging we're adding in. Thanks for the help though! I'm hoping at some point I'll be able to spend some more time diving into elixir/phoenix and may be able to return to this.

**Reply by robinkwilson** [reply]
I know elixir is a new beast. It was a fun challenge to ramp up reading elixir code for me as well. 

I created an Elixir Cheatsheet to help me understand the syntax quickly. 

Maybe this gist can help you too!
https://gist.github.com/robin-k-wilson/5d864656453f0fa80367b931b92a617d

And of course the Elixir Phoenix docs are amazing too: https://elixir-lang.org/getting-started/basic-types.html

**Reply by nguyenvd3** [reply]
Hi @robinkwilson Can you reup your Elixir Cheatsheet because the github link you share has been broken, thanks !

**Comment by agriggs2** [comment, marked as answer]
I am still hitting this issues with latest code, there is no package.json in assets folder (https://github.com/mozilla/reticulum/tree/master/assets) so I am not sure why `npm ci` is needed as it just fails with `ENOENT: no such file or directory, open '/reticulum/assets/package.json'` 

I have been asking for help on discord with no luck, any ideas how to solve in August 2021?
https://discord.com/channels/498741086295031808/535606666708910101/872624114022637568
https://discord.com/channels/498741086295031808/535606666708910101/872897113279836251

**Reply by pattersonbl2** [reply]
We currently don't have a fix for the issue, I will try to reproduce the issue on my local machine to see if i can find a fix but i don't have a time table on when a  possible fix will be found.

**Reply by agriggs2** [reply]
I will keep trying to debug it as well. thanks

**Reply by agriggs2** [reply]
@Pattersonbl2 i keep seeing `[info] TLS :server: In state :abbreviated received CLIENT ALERT: Fatal - Certificate Unknown` prior to the cache timing out.

**Reply by pattersonbl2** [reply]
I have found out we don't have a fix for this.  I will direct you to this discussion for the reasoning: https://github.com/mozilla/hubs/discussions/3648#discussioncomment-263832

**Reply by code-with-qasim** [reply]
I just encountered the same issue. Any Resolution on this?
